net-snmp: add knob whether nlist.h are checked

Previously, it still was checked when there was no nlish.h in sysroots directory.
Add knob to decide whether nlist.h are checked or not.

Upstream-status: Pending

Signed-off-by: Chong Lu <Chong.Lu@windriver.com>
---
 configure.d/config_os_headers |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.d/config_os_headers b/configure.d/config_os_headers
index d903f58..a05d30e 100644
--- a/configure.d/config_os_headers
+++ b/configure.d/config_os_headers
@@ -31,6 +31,7 @@ AC_CHECK_HEADERS([getopt.h   pthread.h  regex.h      ] dnl
                  [sys/un.h           ])
 
 #  Library and Agent:
+if test "x$with_elf" != "xno"; then
 AC_CHECK_HEADERS([nlist.h],,,[
 AC_INCLUDES_DEFAULT
 [
@@ -38,6 +39,7 @@ AC_INCLUDES_DEFAULT
 #define LIBBSD_DISABLE_DEPRECATED 1
 #endif
 ]])
+fi
 
 #  Library:
 AC_CHECK_HEADERS([fcntl.h    io.h       kstat.h                 ] dnl
-- 
1.7.9.5

